skore/src/skore/_sklearn/_plot/data/table_report.py

```diff
@@ -0,0 +1,697 @@
+                    is_x_axis=is_x_axis,
+                    size_per_category=size_per_category,
+                )
+
+        sns.despine(self.figure_, **despine_params)
+
+        self.ax_.set(xlabel=sbd.name(x), ylabel=sbd.name(y))
+        if self.ax_.legend_ is not None:
+            sns.move_legend(self.ax_, (1.05, 0.0))
+
+    def _plot_cramer(self, *, heatmap_kwargs: dict[str, Any] | None) -> None:
+        """Plot Cramer's V correlation among all columns.
+
+        Parameters
+        ----------
+        heatmap_kwargs : dict, default=None
+            Keyword arguments to be passed to seaborn's heatmap.
+        """
+        if heatmap_kwargs is None:
+            heatmap_kwargs = self._default_heatmap_kwargs or {}
+
+        heatmap_kwargs_validated = _validate_style_kwargs(
+            {
+                "xticklabels": True,
+                "yticklabels": True,
+                "robust": True,
+                "square": True,
+                "annot": True,
+                "annot_kws": {"size": 10},
+                "fmt": ".2f",
+            },
+            heatmap_kwargs or {},
+        )
+
+        cramer_v_table = pd.DataFrame(
+            _column_associations._cramer_v_matrix(self.summary["dataframe"]),
+            columns=self.summary["dataframe"].columns,
+            index=self.summary["dataframe"].columns,
+        )
+        # only show the lower triangle of the heatmap since it is a correlation matrix
+        # and keep the diagonal as well.
+        mask = np.triu(np.ones_like(cramer_v_table, dtype=bool), k=1)
+
+        sns.heatmap(cramer_v_table, mask=mask, ax=self.ax_, **heatmap_kwargs_validated)
+        self.ax_.set(title="Cramer's V Correlation")
+
+    def frame(
+        self, *, kind: Literal["dataset", "top-associations"] = "dataset"
+    ) -> pd.DataFrame:
+        """Get the data related to the table report.
+
+        Parameters
+        ----------
+        kind : {'dataset', 'top-associations'}
+            The kind of data to return.
+
+        Returns
+        -------
+        DataFrame
+            The dataset used to create the table report.
+        """
+        if kind == "dataset":
+            return self.summary["dataframe"]
+        elif kind == "top-associations":
+            return pd.DataFrame(self.summary["top_associations"])
+        else:
+            raise ValueError(f"Invalid kind: {kind!r}")
+
+    def _html_repr(self) -> str:
+        """Show the HTML representation of the report."""
+        return to_html(self.summary, standalone=False, column_filters=None)
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}(...)>"
+
+    def _json(self) -> str:
+        """Serialize the data of this report in JSON format.
+
+        It is the serialization chosen to be sent to skore-hub.
+        """
+        to_remove = ["dataframe"]
+        data = {k: v for k, v in self.summary.items() if k not in to_remove}
+        return json.dumps(data, cls=JSONEncoder)
```

Comment:
Why do we keep the sample_table here? It is dropped [in skrub](https://github.com/skrub-data/skrub/blob/main/skrub/_reporting/_table_report.py#L283). It triggers an error when there is timestamp data inside the table.

Reply:
I need to check but I think that `sample_table`, is the preview table shown for the HTML representation and thus the information that Matthieu will need to show. So having them precomputed could be handy but I need to check more in details.

Reply:
So `sample_data` contains the data required to easily build the HTML snippet I think. I think that it could be nice to have @rouk1 in the loop to know what was required in the first version to rebuild the `TableReport` on the front-end.

We could drop it if not required. Otherwise the alternative is to solve the bug and make sure that `Timestamp` value get converted into a string when serializing (in isoformat for instance https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.isoformat.html)

Reply:
Be aware that this function is completely private for the moment and I would say that https://github.com/probabl-ai/skore-hub/issues/424 will defined the most adequate behaviour (it is the point 1 of the checklist). So we might merge without the `_to_json` or make it a TODO with minimal testing.